packages/iov-core/src/signingservercore.ts

```diff
@@ -9,29 +9,57 @@ import { UserProfile, WalletId } from "@iov/keycontrol";
 
 import { MultiChainSigner } from "./multichainsigner";
 
+export interface GetIdentitiesAuthorization {
+  /**
+   * Called by the signing server and lets user decide which of the
+   * available identities should be revealed to the application.
+   */
+  (reason: string, matchingIdentities: ReadonlyArray<PublicIdentity>): Promise<ReadonlyArray<PublicIdentity>>;
+}
+
+export interface SignAndPostAuthorization {
+  /**
+   * Called by the signing server and lets user decide if they want to
+   * authorize a sign and post request
+   */
+  (reason: string, transaction: UnsignedTransaction): Promise<boolean>;
+}
+
 export class SigningServerCore {
   private readonly signer: MultiChainSigner;
   private readonly profile: UserProfile;
+  private readonly authorizeGetIdentities: GetIdentitiesAuthorization;
+  private readonly authorizeSignAndPost: SignAndPostAuthorization;
 
-  constructor(profile: UserProfile, signer: MultiChainSigner) {
+  constructor(
+    profile: UserProfile,
+    signer: MultiChainSigner,
+    authorizeGetIdentities: GetIdentitiesAuthorization,
+    authorizeSignAndPost: SignAndPostAuthorization,
+  ) {
     this.signer = signer;
     this.profile = profile;
+    this.authorizeGetIdentities = authorizeGetIdentities;
+    this.authorizeSignAndPost = authorizeSignAndPost;
   }
 
   public async getIdentities(
-    _: string,
+    reason: string,
     chainIds: ReadonlyArray<ChainId>,
   ): Promise<ReadonlyArray<PublicIdentity>> {
     const matchingIdentities = this.allIdentities().filter(identity => {
       return chainIds.some(chainId => identity.chainId === chainId);
     });
 
-    // TODO: ask user for permission and allow selection of identities
+    const authorizedIdentities = this.authorizeGetIdentities(reason, matchingIdentities);
```

Comment:
So, I assume we can return the whole list or a filtered list for success.

On failure, what is proper, return `[]` or throw and error (reject the promise)?

It would be good to add this to the docs above, what is expected from the callback

Reply:
Whatever the user decides is not an error. Right now, revealing 0 identities means either
- the user rejected the identities request
- the user deselected all available addresses and confirmed

I don't think there is a lot of value in having different return values for those cases.

> It would be good to add this to the docs above, what is expected from the callback

Yes, will improve